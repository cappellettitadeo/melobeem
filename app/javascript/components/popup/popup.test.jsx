import { shallow } from "enzyme";
import React from "react";

import Popup from "./popup";

describe("Components", () => {
  describe("Popup", () => {
    it("should render VotePopup component", () => {
      const mockCallBack = jest.fn();
      const component = shallow(
        <Popup
          handlePopupClose={mockCallBack}
          linkId={1}
          type="vote"
          name="test"
        />
      );

      expect(component).toMatchSnapshot();
    });

    it("should call onClick method", () => {
      const mockCallBack = jest.fn();
      const component = shallow(
        <Popup
          handlePopupClose={mockCallBack}
          linkId={1}
          type="vote"
          name="test"
        />
      );
      expect(mockCallBack.mock.calls.length).toBe(0);
      component.find(".popup__close").simulate("click");
      expect(mockCallBack.mock.calls.length).toBe(1);
    });
  });
});
