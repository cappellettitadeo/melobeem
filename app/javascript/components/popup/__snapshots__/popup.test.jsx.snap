// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Popup should render VotePopup component 1`] = `ShallowWrapper {}`;
